pkg/services/scm_provider/gitlab.go

```diff
@@ -60,6 +61,31 @@ func (g *GitlabProvider) GetBranches(ctx context.Context, repo *Repository) ([]*
 	return repos, nil
 }
 
+func (g *GitlabProvider) GetPullRequests(ctx context.Context, repo *Repository) ([]*Repository, error) {
+	repos := []*Repository{}
+	if !g.allPullRequests {
+		return repos, nil
+	}
+
+	pullRequests, err := g.listPullRequests(ctx, repo)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, pullRequest := range pullRequests {
+		repos = append(repos, &Repository{
+			Organization: repo.Organization,
+			Repository:   repo.Repository,
+			URL:          repo.URL,
+			Branch:       pullRequest.Title,
+			SHA:          pullRequest.SHA,
+			Labels:       pullRequest.Labels,
```

Comment:
What if instead of setting the top-level labels field to be the PR labels, we added a `PullRequest` field with fields such as `SourceBranch`, `TargetBranch`, and `Labels`? That would leave the top-level labels field to have its usual meaning (repo labels).

The `PullRequest` field should also have an `ID` field to match what's provided by the pull request generator.

Reply:
This is a very good suggestion, I'll implement it. Also, this way we get to keep the dedicated `if` statement for `filter.PullRequestLabelMatch` in `matchFilter` function.